components/content-service/pkg/storage/minio.go

```diff
@@ -364,6 +364,36 @@ func (s *presignedMinIOStorage) SignUpload(ctx context.Context, bucket, obj stri
 	return &UploadInfo{URL: url.String()}, nil
 }
 
+func (s *presignedMinIOStorage) DeleteObject(ctx context.Context, bucket string, query *DeleteObjectQuery) (err error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "minio.DeleteObject")
+	defer tracing.FinishSpan(span, &err)
+
+	if query.Name != "" {
+		err = s.client.RemoveObject(ctx, bucket, query.Name, minio.RemoveObjectOptions{})
+		if err != nil {
+			log.WithField("bucket", bucket).WithField("object", query.Name).Error(err)
+			return err
+		}
+	}
+	if query.Prefix != "" {
+		objectsCh := make(chan minio.ObjectInfo)
+		go func() {
+			defer close(objectsCh)
+			for object := range s.client.ListObjects(ctx, bucket, minio.ListObjectsOptions{
+				Prefix:    query.Prefix,
+				Recursive: true,
+			}) {
+				objectsCh <- object
+			}
+		}()
+		for removeErr := range s.client.RemoveObjects(ctx, bucket, objectsCh, minio.RemoveObjectsOptions{}) {
```

Comment:
is `removeErr == nil` at the end of the delete operation?

Reply:
At least behaves like that, you can trigger if by going to `Synched User Data` view and cleaning all data. Then go to https://minio.ak-code-user-data-sync.staging.gitpod-dev.com and check whether previous user data were removed. It works for me.

It's also done like that in minio docs: https://docs.min.io/docs/golang-client-api-reference.html#RemoveObjects

Reply:
note to self: `RemoveObjects` returns a channel which gets closed. Hence the `nil` case doesn't have to be handled.